### PR TITLE
feat: specify labels that prevent PR auto-merges

### PR DESCRIPTION
**Pull request by Chriscbr**
Allow specifying labels that prevent PR auto-merges through the `withoutLabels` field in the `AutoMerge` component.

BREAKING CHANGE: `AutoMerge` now expects a `GitHub` component as its first argument, and no longer requires `Mergify` to be provided.

---
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .projenrc.js [hunk 1]
```diff
@@ -119,7 +121,15 @@ project.vscode.launchConfiguration.addConfiguration({
...
-project.github.mergify.addRule({
+const mergify = new github.Mergify(project.github);
+
+new github.AutoMerge(project, {
+  mergify: mergify,
+  buildJob: project.buildWorkflowJobId,
+  withoutLabels: ['do-not-merge'],
+});
+
+mergify.addRule({
```
Comment (Chriscbr) [inline .projenrc.js:132]: Wasn't sure if I should expose the new option through some other field within the project constructor, like `autoMergeOptions` or something....

File: src/github/auto-merge.ts [hunk 2]
```diff
@@ -18,6 +18,12 @@ export interface AutoMergeOptions {
    * @default 1
    */
   readonly approvedReviews?: number;
+
+  /**
+   * List of labels that will prevent merging.
+   * @default []
+   */
+  readonly withoutLabels?: string[];
```
Comment (Chriscbr) [inline src/github/auto-merge.ts:26]: Not sure if I love this name - maybe something else like  `blockingLabels` would be better?

File: .projenrc.js (hunk 1 above)
Comment (eladb) [reply .projenrc.js:132]: Yes, I think `autoMergeOptions` is called for

File: src/github/auto-merge.ts (hunk 2 above)
Comment (eladb) [reply src/github/auto-merge.ts:26]: Yes, this name is not ideal... `blockingLabels` is better.

File: src/github/auto-merge.ts [hunk 3]
```diff
@@ -18,6 +18,12 @@ export interface AutoMergeOptions {
    * @default 1
    */
   readonly approvedReviews?: number;
+
+  /**
+   * List of labels that will prevent merging.
+   * @default []
```
Comment (eladb) [inline src/github/auto-merge.ts:24]: I am comfortable with `do-not-merge` as a default blocking label

**Review by eladb — CHANGES_REQUESTED** [review]

**Review by eladb — APPROVED** [review]

**Comment by eladb** [comment]
BTW @Chriscbr you should be able to push branches to this repo directly now and not to a fork.

**Pull request merged**